public/app/features/dashboard/api/v2.ts

```diff
@@ -62,7 +69,59 @@ export class K8sDashboardV2API implements DashboardAPI<DashboardWithAccessInfo<D
     throw new Error('Method not implemented.');
   }
 
-  saveDashboard(options: SaveDashboardCommand): Promise<SaveDashboardResponseDTO> {
-    throw new Error('Method not implemented.');
+  async saveDashboard(options: SaveDashboardCommand<DashboardV2Spec>): Promise<SaveDashboardResponseDTO> {
+    const dashboard = options.dashboard;
+
+    const obj: ResourceForCreate<DashboardV2Spec> = {
+      // the metadata will have the name that's the uid
+      metadata: {
+        ...options?.k8s,
+      },
+      spec: {
+        ...dashboard,
+      },
+    };
+
+    // add annotations
+    if (options.message) {
+      obj.metadata.annotations = {
+        ...obj.metadata.annotations,
+        [AnnoKeyMessage]: options.message,
+      };
+    } else if (obj.metadata.annotations) {
+      delete obj.metadata.annotations[AnnoKeyMessage];
+    }
+
+    // add folder annotation
+    if (options.folderUid) {
+      obj.metadata.annotations = {
+        ...obj.metadata.annotations,
+        [AnnoKeyFolder]: options.folderUid,
+      };
+    }
+
+    if (obj.metadata.name) {
+      return this.client.update(obj).then((v) => this.asSaveDashboardResponseDTO(v));
+    }
+    return await this.client.create(obj).then((v) => this.asSaveDashboardResponseDTO(v));
+  }
+
+  asSaveDashboardResponseDTO(v: Resource<DashboardV2Spec>): SaveDashboardResponseDTO {
+    const url = locationUtil.assureBaseUrl(
+      getDashboardUrl({
+        uid: v.metadata.name,
+        currentQueryParams: '',
+        slug: kbn.slugifyForUrl(v.spec.title),
+      })
+    );
+
+    return {
+      uid: v.metadata.name,
+      version: v.spec.version ?? 0, // FIXME: what is version?
+      id: v.spec.id ?? 0,
```

Comment:
`spec.id ` is no more, this should be taken from `metadata.labels["grafana.app/deprecatedInternalID"]` (ref https://github.com/grafana/grafana/pull/98311), right @ryantxu ?

Think we should remove the id from the spec altogether.

Reply:
Fixed in [`b0aadbc` (#98263)](https://github.com/grafana/grafana/pull/98263/commits/b0aadbc8301ecfdf97eadb8e7fbe14a39f2c6f89)

Reply:
🙏